packages/realm/src/Object.ts

```diff
@@ -232,7 +232,7 @@ export class RealmObject<T = DefaultObject, RequiredProperties extends keyof Omi
           result[propertyName] = propertyValue;
         }
       } else {
-        if (typeof defaultValue !== "undefined") {
+        if (created && typeof defaultValue !== "undefined") {
```

Comment:
Since there's also a `created` guard on line 240, I wonder if it would be better to turn the entire `else` block (line 234) into an `else if (created)` block, instead.